### PR TITLE
Added the maven profile to build rpm packages

### PR DESCRIPTION
**Pull request by gaojun2048**
## What is the purpose of the pull request
**Hello , This pull request adds rpmbuild plugin**

**The associated issue is Develops an Ambari plugin about DS : https://github.com/apache/incubator-dolphinscheduler/issues/1469**

## Brief change log

*add rpmbuild plugin*
  - *Add maven-dependency-plugin and rpm-maven-plugion in root pom.xml*

  - *Add profile rpmbuild in dolphinscheduler-dist/pom.xml*

  - *Removed some duplicate operations in dolphinscheduler-dist/src/main/assembly/dolphinscheduler-binary.xml *

### CODE REVIEW

**Comment by gaojun2048** [comment]
> /opt/soft dir will auto create or not ?
> 
> by the way, how to build rpm package？

1、The build command:
mvn -U clean package -Prpmbuild -Dmaven.test.skip=true

2、Sorry , I just add /opt/soft auto create code

**Comment by gaojun2048** [comment]
@dailidong   I added auto create /opt/soft code

File: dolphinscheduler-dist/pom.xml [hunk 1]
```diff
@@ -101,6 +101,276 @@
...
+
+                            <defineStatements>
+                                <!-- disable compile python when rpm build -->
+                                <defineStatement>__os_install_post %(echo '%{__os_install_post}' | sed -e 's!/usr/lib[^[:space:]]*/brp-python-bytecompile[[:space:]].*$!!g')</defineStatement>
+                            </defineStatements>
+                            <mappings>
+                                <mapping>
+                                    <directory>/opt/soft/${project.build.finalName}/conf</directory>
+                                    <filemode>755</filemode>
+                                    <username>root</username>
```
Comment (lgcareer) [inline dolphinscheduler-dist/pom.xml]: I see the user and group is root default,if other user run `mvn clean package -P rpmbuild`,is it ok?

Comment (gaojun2048) [reply dolphinscheduler-dist/pom.xml]: Yes, It`s ok .
I tested build with username argo , build success

Comment (gaojun2048) [reply dolphinscheduler-dist/pom.xml]: > I see the user and group is root default,if other user run `mvn clean package -P rpmbuild`,is it ok?

Yes, It`s ok .
I tested build with username argo , build success

**Comment by dailidong** [comment]
please add  how to use document on official website github : https://github.com/apache/incubator-dolphinscheduler-website

**Pull request merged**